### PR TITLE
Docker environment for testing the new metadata templates

### PR DESCRIPTION
**Pull request by PaulVanSchayck**
This builds a complete docker environment (two containers) to run the irods_avu_json work. Based  from the MaastrichtUniversity/irods_avu_json-docker repository.

@trel Did we also suggest to merge the MaastrichtUniversity/irods_avu_json-ruleset repo into here. Does that make sense?

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+The repository [irods_avu_json-ruleset](https://github.com/MaastrichtUniversity/irods_avu_json) contains the iRODS rules, policies and microservices to make the conversion code operational in iRODS.
+
+This docker-compose binds everything together in a docker example.
+
+## Running
+This will spawn the iCAT container and a postgres DB container.
+```
+docker-compose build
+docker-composer up
```
Comment (trel) [inline docker/README.md]: "docker-composer" -> "docker-compose"

File: docker/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+This is a docker environment to test the [irods_avu_json](https://github.com/MaastrichtUniversity/irods_avu_json) Python 
+module. Please read the README of that repository for an introduction into the goals of this work.
+
+The repository [irods_avu_json-ruleset](https://github.com/MaastrichtUniversity/irods_avu_json) contains the iRODS rules, policies and microservices to make the conversion code operational in iRODS.
+
+This docker-compose binds everything together in a docker example.
+
+## Running
+This will spawn the iCAT container and a postgres DB container.
```
Comment (trel) [inline docker/README.md]: "iCAT" -> "catalog provider"

File: docker/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+version: '2'
+services:
+  irods:
```
Comment (trel) [inline docker/docker-compose.yml]: "irods" -> "irods-provider"

File: docker/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+docker-composer up
+```
+
+## Testing
+A small test script is available in `/irods_avu_json-ruleset/tests/pep_tests.sh`. This script throws lots of errors
+and warnings, but should report when a unexpected error happens.
+
+You can run it using
+```
+docker exec -it irods_avu_json-docker_irods_1 bash
```
Comment (trel) [inline docker/README.md]: "irods_avu_json-docker_irods_1" -> "irods_avu_json-docker_irods-provider_1"

File: docker/docker/bootstrap.sh [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    # set up the iCAT database
+    /opt/irods/setupdb.sh /etc/irods/setup_responses
+
+    # set up iRODS
+    python /var/lib/irods/scripts/setup_irods.py < /etc/irods/setup_responses
+
+    # Add python rule engine to iRODS
+    /opt/irods/add_rule_engine.py /etc/irods/server_config.json python 1
+
+    touch /var/run/irods_installed
```
Comment (trel) [inline docker/docker/bootstrap.sh]: can it just use `/var/lib/irods/VERSION.json` instead?    here and on line 29?

File: docker/docker/setup_responses [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+1
+1
+irods-db
+5432
+icat
+irods
+yes
+pgB3rKhqLGFLora
+tempsalt
+nlmumc
```
Comment (trel) [inline docker/docker/setup_responses]: is this what we want?

**Comment by PaulVanSchayck** [comment]
Hi @trel. Thanks for your comments. I've done some cleaning to the code, and verified everything still works.

**Comment by trel** [comment]
looks great - thanks!

**Pull request merged**